### PR TITLE
Add generate-pending-release-diffs command

### PR DESCRIPTION
**Pull request by westonruter**
To ensure that we're only including the expected files when we do releases, it's important to review a diff of the changes for each plugin prior to doing a release. This PR introduces a `generate-pending-release-diffs` command to facilitate this. 

See example output on https://github.com/WordPress/performance/pull/1225#issuecomment-2118460437

You can invoke via:

```
npm run generate-pending-release-diffs --silent > overview.md
```

And then the contents can be pasted into a GitHub comment for review. Note the `--silent` arg to `npm run` omits this output:

```
> generate-pending-release-diffs
> bin/generate-pending-release-diffs.sh


> build:plugin:performance-lab
> rm -rf build/performance-lab && mkdir -p build/performance-lab && git archive HEAD | tar -x -C build/performance-lab


> build-plugins
> npm-run-all 'build:plugin:!(performance-lab)'


> build:plugin:auto-sizes
> webpack --mode production --env plugin=auto-sizes
```

Or you can just directly invoke `./bin/generate-pending-release-diffs.sh`. 

On MacOS, you can put the output directly onto the clipboard via:

```sh
./bin/generate-pending-release-diffs.sh | pbcopy
```

On Linux, there are equivalent commands.

### CODE REVIEW

**Review by thelovekesh — APPROVED** [review]

**Comment by thelovekesh** [comment]
It would have been required as part of https://github.com/WordPress/performance/issues/1061. Great that it has been added by @westonruter 🎉

File: bin/generate-pending-release-diffs.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+done
+
+cd "$(git rev-parse --show-toplevel)"
+
+npm run build:plugin:performance-lab >&2 # TODO: Remove this as of <https://github.com/WordPress/performance/pull/1182>.
+npm run build-plugins >&2
+
+stable_dir=/tmp/stable-svn
+mkdir -p "$stable_dir"
+for plugin_slug in $(jq '.plugins[]' -r plugins.json) 'performance-lab'; do
```
Comment (westonruter) [inline bin/generate-pending-release-diffs.sh]: Remove with merging #1182.

```suggestion
for plugin_slug in $(jq '.plugins[]' -r plugins.json); do
```

File: bin/generate-pending-release-diffs.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+for required_command in npm git svn jq rsync; do
+	if ! command -v "$required_command" &> /dev/null; then
+		echo "Error: The $required_command command must be installed to use this script." >&2
+		exit 1
+	fi
+done
+
+cd "$(git rev-parse --show-toplevel)"
+
+npm run build:plugin:performance-lab >&2 # TODO: Remove this as of <https://github.com/WordPress/performance/pull/1182>.
```
Comment (westonruter) [inline bin/generate-pending-release-diffs.sh]:
```suggestion
```

**Comment by westonruter** [comment]
Current output:

# `auto-sizes`

`svn status`:
```
```

<details><summary><code>svn diff</code></summary>

```diff
```
</details>

# `dominant-color-images`

`svn status`:
```
M       class-dominant-color-image-editor-imagick.php
```

<details><summary><code>svn diff</code></summary>

```diff
Index: class-dominant-color-image-editor-imagick.php
===================================================================
--- class-dominant-color-image-editor-imagick.php	(revision 3091115)
+++ class-dominant-color-image-editor-imagick.php	(working copy)
@@ -32,7 +32,6 @@
 		}
 
 		try {
-			// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
 			// The logic here is resize the image to 1x1 pixel, then get the color of that pixel.
 			$this->image->resizeImage( 1, 1, Imagick::FILTER_LANCZOS, 1 );
 			$pixel = $this->image->getImagePixelColor( 0, 0 );
```
</details>

# `embed-optimizer`

`svn status`:
```
```

<details><summary><code>svn diff</code></summary>

```diff
```
</details>

# `optimization-detective`

`svn status`:
```
```

<details><summary><code>svn diff</code></summary>

```diff
```
</details>

# `performance-lab`

`svn status`:
```
!       LICENSE
```

<details><summary><code>svn diff</code></summary>

```diff
```
</details>

# `speculation-rules`

`svn status`:
```
M       helper.php
M       load.php
```

<details><summary><code>svn diff</code></summary>

```diff
Index: helper.php
===================================================================
--- helper.php	(revision 3091115)
+++ helper.php	(working copy)
@@ -103,7 +103,7 @@
 					// Also exclude rel=nofollow links, as plugins like WooCommerce use that on their add-to-cart links.
 					array(
 						'not' => array(
-							'selector_matches' => 'a[rel=nofollow]',
+							'selector_matches' => 'a[rel~="nofollow"]',
 						),
 					),
 				),
Index: load.php
===================================================================
--- load.php	(revision 3091115)
+++ load.php	(working copy)
@@ -5,7 +5,7 @@
  * Description: Enables browsers to speculatively prerender or prefetch pages when hovering over links.
  * Requires at least: 6.4
  * Requires PHP: 7.2
- * Version: 1.3.0
+ * Version: 1.3.1-alpha
  * Author: WordPress Performance Team
  * Author URI: https://make.wordpress.org/performance/
  * License: GPLv2 or later
@@ -65,7 +65,7 @@
 	}
 )(
 	'plsr_pending_plugin_info',
-	'1.3.0',
+	'1.3.1-alpha',
 	static function ( string $version ): void {
 
 		// Define the constant.
```
</details>

# `webp-uploads`

`svn status`:
```
```

<details><summary><code>svn diff</code></summary>

```diff
```
</details>

**Comment by westonruter** [comment]
@thelovekesh I see the `LICENSE` file is now missing in the latest `performance-lab` build here. I suppose that is correct because the file is really meant for the repo and not for the individual plugins, and this file is absent from the other plugins as well.

**Pull request merged**